airflow/utils/task_group.py

```diff
@@ -19,11 +19,25 @@
 A TaskGroup is a collection of closely related tasks on the same DAG that should be grouped
 together when the DAG is displayed graphically.
 """
-
-from typing import TYPE_CHECKING, Dict, Generator, List, Optional, Sequence, Set, Union
+import functools
+import re
+from typing import (
+    TYPE_CHECKING,
+    Callable,
+    Dict,
+    Generator,
+    List,
+    Optional,
+    Sequence,
+    Set,
+    TypeVar,
+    Union,
+    cast,
+)
 
 from airflow.exceptions import AirflowException, DuplicateTaskIdFound
 from airflow.models.taskmixin import TaskMixin
+from airflow.utils.decorators import signature
```

Comment:
```suggestion
from inspect import signature
```

Please import directly form the module, not from somewhere else where we us it.